src/backend/storage/file/buffile.c

```diff
@@ -490,6 +490,8 @@ BufFileRead(BufFile *file, void *ptr, size_t size)
 	size_t		nread = 0;
 	size_t		nthistime;
 
+	Insist(file);
```

Comment:
I think you should replace condition https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/executor/execHHashagg.c#L1881 by assert

Reply:
Done

Reply:
I think this insist is redundant

Reply:
I still think so

Reply:
Done